lab4/provided/record_store.hh

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+class Record {
+public:
+    int64_t ticksFromNow;
+    bool write;
+    uint64_t address;
+    int requestId;
+    int size;
+    vector<uint8_t> dataVec;
+
+    Record(int64_t ticks = 5, bool wr = false, uint64_t addr = 0x10000, int reqId = 0, int size = 4, vector<uint8_t> data = {}):
```

Comment:
Rather than having sane (weird) defaults, I think it's better to have no parameters to the constructor and instead use all 0 for the data given to the members below.

Reply:
The constructor has actually proven useful for writing small scale tests. I can also replace the defaults with 0/null instead, to get similar behavior, if this is problematic.